design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -941,3 +1001,154 @@ Libvirt configuration snippet:
     <rom enabled='no'/>
 </interface>
 ```
+
+## Appendix F: Concrete example for vDPA binding
+
+Present a concrete example of the actions needed at each integration
+point, considering the option suggested in the
+[design section](#design-details).
+
+### Kubevirt CR Specification
+
+The vDPA binding is registered to the Kubevirt CR.
+The plugin is then referenced by name from the VM Specification.
+
+> **Note**: The vDPA binding does not need to perform changes in the
+> pod network, therefore, there is no need for a
+> network-attachment-definition.
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    network:
+      binding:
+        vdpa:
+          sidecarImage: quay.io/kubevirt/network-vdpa-binding
+          downwardAPI: device-info
+```
+
+### VM Specification
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vmi-1-bridge
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: blue
+              mac: 12:34:56:78:9a:bc
+              binding:
+                name: vdpa
+      networks:
+        - name: blue
+          multus:
+            networkName: vdpa-network
+```
+
+### NetworkAttachmentDefinition Specification
+
+Not required for vDPA binding
+
+> **Note**: Defining a NetworkAttachmentDefinition to connect the pod to the node vDPA interface is required,
+> this NetworkAttachmentDefinition is specified in the VM spec.
+> No additional NetworkAttachmentDefinition is required, because there is no need to configure the pod network
+> stack for vDPA.
+
+### Webhook Admitter
+
+- The core KubeVirt admitter assures that only one method of binding
+  definition is possible: Either through the Interface API or Network Binding Plugin API.
+- No addmiter is provided as part of the vDPA binding.
+
+### Pod Definition
+
+The VMI controller updates the pod manifest per the VMI specification.
+
+- From the VMI spec, the following snippet is detected:
+
+```yaml
+name: blue
+binding:
+  name: vdpa
+```
+
+- As a result, the pod is updated with the following annotations & spec:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: virt-launcher-123
+  annotations:
+    kubevirt.io/network-info: '{
```

Comment:
If the same information is available as a different annotation `k8s.v1.cni.cncf.io/network-status`, out of curiosity what problem does having a kubevirt specific annotation solve?

https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/how-to-use.md#launch-pod-with-text-annotation

Reply:
The network-status annotation contains information about all the pod interfaces, including IPs and other confidential data.The new annotation collects from the status only the device info of the interfaces that use a plug-in that supports it, the other interface and other info besides the device-infi is not passed.
We find this approach more secure and avoiding sidecar users to get the complete pod networking information.

Reply:
Done, added it to the alternatives sections.